services/wallet/common/log_parser.go

```diff
@@ -251,52 +247,153 @@ func ParseErc721TransferLog(ethlog *types.Log) (from, to common.Address, tokenID
 		log.Warn("not enough topics for erc721 transfer", "topics", ethlog.Topics)
 		return
 	}
-	if len(ethlog.Topics[1]) != 32 {
-		log.Warn("second topic is not padded to 32 byte address", "topic", ethlog.Topics[1])
+
+	var err error
+	from, to, err = getFromToAddresses(*ethlog)
+	if err != nil {
+		log.Error("log_parser::ParseErc721TransferLog", err)
 		return
 	}
-	if len(ethlog.Topics[2]) != 32 {
-		log.Warn("third topic is not padded to 32 byte address", "topic", ethlog.Topics[2])
+	tokenID.SetBytes(ethlog.Topics[3][:])
+
+	return
+}
+
+func GetLogSubTxID(log types.Log) common.Hash {
+	// Get unique ID by using TxHash and log index
+	index := [4]byte{}
+	binary.BigEndian.PutUint32(index[:], uint32(log.Index))
+	return crypto.Keccak256Hash(log.TxHash.Bytes(), index[:])
+}
+
+func getLogSubTxIDWithTokenID(log types.Log, tokenID uint16) common.Hash {
```

Comment:
hmm can't the same tokenID appear twice in the list of ids for a `TransferBatch`? That would result in the same subTxID for multiple transfer entries. I'd use the corresponding index of the `ids` array instead to make sure the hash is unique.